### ISSUE TITLE
Errors from custom providers are rendered in cli but not web UI

### ISSUE THREAD

**Issue by aantn** [promptfoo/promptfoo#988]
When using a custom provider, errors show up in the cli:

<img width="1387" alt="Screenshot 2024-06-22 at 19 23 24" src="https://github.com/promptfoo/promptfoo/assets/494087/8f411b07-691f-4b4d-9201-3a0c1a2c1fd9">

But in the UI the test is marked as failure with no clear reason:

<img width="1165" alt="Screenshot 2024-06-22 at 19 21 50" src="https://github.com/promptfoo/promptfoo/assets/494087/f2f97034-4bcc-4f09-b95b-1e8952307a60">

It would be great to see the same error in the UI.

Here is a minimal reproducible example:

promptfoo config:
```
prompts:
  - "foo"

providers:
  - id: 'python:./test.py'

tests:
  - vars:
      x: y
    description: abc
```

python provider (test.py):
```
def call_api(prompt, options, context):
    return {
        "error": "something went wrong"
    }
```

**Comment by typpo** [comment]
Fix in #990.  Thanks @aantn

**Comment by aantn** [comment]
Thank you very much as always.

**Comment by typpo** [comment]
Released in 0.66.0
